### PR TITLE
Add `MultiRZ` to the list of supported gates

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
This PR is based on #28 and adds `MultiRZ`'s gate and generator with a direct use of cuQuantum.

### CODE REVIEW

File: pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp [hunk 1]
```diff
@@ -601,6 +607,20 @@ class StateVectorCudaManaged
...
+    inline void applyMultiRZ(const std::vector<std::size_t> &wires,
+                             bool adjoint, Precision param) {
+        static const std::vector<std::string> names(wires.size(), {"RZ"});
+        applyParametricPauliGate(names, {}, wires, param, adjoint);
+    }
+    inline void applyGeneratorMultiRZ(const std::vector<std::size_t> &wires,
+                                      bool adjoint) {
+        static const std::string name{"PauliZ"};
+        static const Precision param = 0.0;
+        applyDeviceMatrixGate(gate_cache_.get_gate_device_ptr(name, param), {},
```
Comment (chaeyeunpark) [inline pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:620]: Doesn't `applyDeviceMatrixGate` need a matrix with size $2^{\rm nTargets} \times 2^{\rm nTargets}$ as in https://docs.nvidia.com/cuda/cuquantum/custatevec/api/functions.html#custatevecapplymatrix ?

**Review by chaeyeunpark — APPROVED** [review]
Looks good to me! Still, I have one single question.

File: pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp (hunk 1 above)
Comment (maliasadi) [reply pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp]: Thank you @chaeyeunpark! You're right. It shouldn't be this; it's been fixed while testing but I forgot to commit it here.

File: pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp [hunk 2]
```diff
@@ -170,7 +174,8 @@ template <class T = double> class AdjointJacobianGPU {
         {"DoubleExcitaitonPlus",
```
Comment (AmintorDusko) [inline pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp:174]:
```suggestion
        {"DoubleExcitationPlus",
```

**Comment by AmintorDusko** [comment]
Hi @maliasadi, nice work!

**Comment by mlxd** [comment]
For non-main branches, feel free to merge freely ;)

**Review by mlxd — APPROVED** [review]

**Pull request merged**